presto-orc/src/main/java/com/facebook/presto/orc/ByteArrayUtils.java

```diff
@@ -83,4 +83,20 @@ private static long longBytesToLong(long bytes)
     {
         return Long.reverseBytes(bytes) ^ Long.MIN_VALUE;
     }
+
+    public static long hash(byte[] bytes, int offset, int length)
```

Comment:
Does this hash algorithm has a name? If so, let's move this method into a separate class named after that algorithm. Otherwise, where does this algorithm come from?